### PR TITLE
Separation of checkAttributeSyntax from semantics in Facility modules

### PR DESCRIPTION
**Pull request by metodej**
- former checkAttributeValue is now separated into checkAttributeSyntax and checkAttributeSemantics in facility attribute modules
- also added test for checks

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_accountExpirationTime.java [hunk 1]
```diff
@@ -22,10 +22,9 @@
 public class urn_perun_facility_attribute_def_def_accountExpirationTime extends FacilityAttributesModuleAbstract implements FacilityAttributesModuleImplApi {
 
 	@Override
-	public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongAttributeValueException {
-		Integer accountExpTime = (Integer) attribute.getValue();
-		if(accountExpTime == null) {
-			throw new WrongAttributeValueException("account expiration time shouldn't be null");
+	public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongReferenceAttributeValueException {
+		if(attribute.valueAsInteger() == null) {
+			throw new WrongReferenceAttributeValueException(attribute, "account expiration time shouldn't be null");
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_accountExpirationTime.java]: Add object facility please.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeMountPoint_passwd_scp.java [hunk 2]
```diff
@@ -21,14 +22,19 @@
...
 
-		if(value == null) throw new WrongAttributeValueException(attribute, "Value can't be null");
+		if (value == null) return;
 		Matcher matcher = pattern.matcher(value);
 		if (!matcher.matches()) throw new WrongAttributeValueException(attribute, "Wrong format. ^(/[-_.A-z0-9]+)+$ expected.");
 	}
 
+	@Override
+	public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility facility, Attribute attribute) throws WrongReferenceAttributeValueException {
+		if (attribute.valueAsString() == null) throw new WrongReferenceAttributeValueException(attribute, "Value can't be null");
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeMountPoint_passwd_scp.java]: Add object facility please.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java [hunk 3]
```diff
@@ -40,6 +40,11 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility faci
 		if(!success) throw new WrongAttributeValueException(attribute, "There is no such facility with the same name like this pbsServer");
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java]: I think this whole part is semantic. And the error is that no core attribute "name" for object facility exists in perun.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_accountExpirationTime.java (hunk 1 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_accountExpirationTime.java]: Facility object added.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeMountPoint_passwd_scp.java (hunk 2 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_homeMountPoint_passwd_scp.java]: Facility object added.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java (hunk 3 above)
Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java]: Okay, the check was moved to semantic part and the error message was changed.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java [hunk 4]
```diff
@@ -37,7 +33,7 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Facility faci
 				break;
 			}
 		}
-		if(!success) throw new WrongAttributeValueException(attribute, "There is no such facility with the same name like this pbsServer");
+		if(!success) throw new WrongReferenceAttributeValueException(attribute, null, facility, null, "No core attribute " + pbsServer + " exist for facility object in perun.");
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_def_pbsServer.java]: This exception is still not what I meant. You should say that there is no facility with the "urn:..facility:..def:core:name" same as the pbsServer attribute. That's what I meant by core attribute.

**Pull request merged**